jenkins/image_building/node-image-cloud-init/retrieve.configuration.files.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+set -e
+url="${1}"
+dst="${2}"
+filename="$(basename "${url}")"
+tmpfile="/tmp/${filename}"
+curl -sSL -w "%{http_code}" "${url}" | sed "s:/usr/bin:/usr/local/bin:g" > /tmp/"${filename}"
```

Comment:
`--fail` is your friend here, all the DIY error handling here seems unnecessary.